packages/core/strapi/lib/commands/opt-out-telemetry.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const { resolve } = require('path');
+const fse = require('fs-extra');
+const chalk = require('chalk');
+const fetch = require('node-fetch');
+const machineID = require('../utils/machine-id');
+
+const readPackageJSON = async path => {
+  try {
+    const packageObj = await fse.readJson(path);
+    const uuid = packageObj.strapi.uuid;
+
+    return [uuid, packageObj];
+  } catch (err) {
+    console.error(`${chalk.red('Error')}: ${err.message}`);
+  }
+};
+
+const writePackageJSON = async (path, file, spacing) => {
+  try {
+    await fse.writeJson(path, file, { spaces: spacing });
+    return true;
+  } catch (err) {
+    console.error(`${chalk.red('Error')}: ${err.message}`);
+  }
+};
+
+const sendEvent = async uuid => {
+  try {
+    await fetch('https://analytics.strapi.io/track', {
+      method: 'POST',
+      body: JSON.stringify({
+        event: 'didOptOutTelemetry',
+        uuid,
+        deviceId: machineID(),
+      }),
+      headers: { 'Content-Type': 'application/json' },
+    });
+  } catch (e) {
+    //...
+  }
+};
+
+module.exports = async function optOutTelemetry() {
+  const packageJSON = resolve(process.cwd(), 'package.json');
+
+  if (!packageJSON) {
```

Comment:
this doesn't check if the package.json file exists this is just a path here. you need to check with fs-extra and `pathExists`

Reply:
There is no such check in this file. Instead, the command is invoked from `packages/core/strapi/bin/strapi.js` and there is this check implemented, which prevents the command from running if no `package.json` is present, or if `package.json` is not a strapi one:
```javascript
const checkCwdIsStrapiApp = name => {
  let logErrorAndExit = () => {
    console.log(
      `You need to run ${yellow(
        `strapi ${name}`
      )} in a Strapi project. Make sure you are in the right directory.`
    );
    process.exit(1);
  };

  try {
    const pkgJSON = require(process.cwd() + '/package.json');
    if (!_.has(pkgJSON, 'dependencies.@strapi/strapi')) {
      logErrorAndExit(name);
    }
  } catch (err) {
    logErrorAndExit(name);
  }
};

Reply:
👍

Reply:
But wouldn't yarn/npm throw an error regardless if `package.json` was not in place? It's not going to run anything since `strapi` command is also going to be unavailable, isn't it?